subprojects/core/src/main/java/org/gradle/internal/service/scopes/DefaultGradleUserHomeScopeServiceRegistry.java

```diff
@@ -117,25 +121,27 @@ public Optional<ServiceRegistry> getCurrentServices() {
     public void release(ServiceRegistry registry) {
         lock.lock();
         try {
-            for (Map.Entry<File, Services> entry : servicesForHomeDir.entrySet()) {
+            release(servicesForHomeDir.entrySet().stream()
```

Comment:
🤔 I wonder if there are performance implications caused by switching to streams.

Reply:
Might be. We should check hard facts. Thanks for testing.

- https://medium.com/@ezzeddinebilel/java-stream-and-for-loop-a-balanced-view-on-performance-a38e45f271e5
- https://wiki.c2.com/?RulesOfOptimization